### PR TITLE
[#175555319] Add function GetBPDTransactions

### PR DESCRIPTION
**Pull request by BurnedMarshal**
Read Citizen's transactions from BPD (pivotal story [#175555319](https://www.pivotaltracker.com/story/show/175555319)
This function implements the following feature:

Model of Transaction using TypeORM (binding with a view transaction)
Fix Database reconnection when connection lost happens
OpenAPI specs for GETBPDTransaction
Database fixtures for local execution

For now this function doesn't have authentication that will be added later with another PR

### CODE REVIEW

File: utils/database.ts [hunk 1]
```diff
@@ -22,12 +23,18 @@ export async function getConnection(
 ): Promise<Connection> {
   // Use an existing connection inside the connection manager.
   const maybeActiveConnection = getConnectionManager().connections[0];
+  if (maybeActiveConnection && maybeActiveConnection.isConnected) {
+    return maybeActiveConnection;
+  }
   if (maybeActiveConnection) {
+    // If the existing connection is disconnected
+    // we perorm a reconnection and return it
```
Comment (gunzip) [inline utils/database.ts:31]:
```suggestion
    // we perform a reconnection and return it
```

Comment (BurnedMarshal) [reply utils/database.ts]: Fixed in 6163fde

File: GetBPDTransactions/handler.ts [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+  transactionRepository: TaskEither<Error, Repository<Transaction>>
+): IHttpHandler {
+  return async (context, requestFiscalCode) => {
+    if (isNone(requestFiscalCode)) {
+      return ResponseErrorValidation(
+        "Bad request",
+        "Missing fiscal code header"
+      );
+    }
+    return transactionRepository
```
Comment (AleDore) [inline GetBPDTransactions/handler.ts:68]: maybe we can chain also this first check with the others?

File: utils/database.ts [hunk 3]
```diff
@@ -22,14 +23,22 @@ export function getConnection(
...
     return taskEither.of(maybeActiveConnection);
   }
+  if (maybeActiveConnection) {
+    // If the existing connection is disconnected
+    // we perform a reconnection and return it
+    return tryCatch(
+      () => maybeActiveConnection.connect(),
+      err => new Error(`Error during postgres re-connection [err:${err}]`)
+    );
+  }
```
Comment (AleDore) [inline utils/database.ts:36]: maybe we can refactor this and collapse into a single chain :)

File: GetBPDTransactions/handler.ts (hunk 2 above)
Comment (BurnedMarshal) [reply GetBPDTransactions/handler.ts]: Fixed in 428e753

File: utils/database.ts (hunk 3 above)
Comment (BurnedMarshal) [reply utils/database.ts]: Fixed in df37e49

**Review by AleDore — APPROVED** [review]
Lgtm.

**Pull request merged**